main.php

```diff
@@ -90,11 +47,13 @@ public static function main(): void
             $word = trim(readline('Enter a word to be hyphenated (leave empty to skip this step): '));
 
             if(strlen($word) > 0) {
-                $dbSourceExecutor = new DatabaseHyphenationExecutor($wordRepository, $ruleRepository, $processor, $word);
-                $fileSourceExecutor = new FileHyphenationExecutor($fileHyphenator, $word);
+                $databaseSourceExecutor = $manager->resolve('DatabaseHyphenationExecutor');
+                $databaseSourceExecutor->setWord($word);
+                $fileSourceExecutor = $manager->resolve('FileHyphenationExecutor');
+                $fileSourceExecutor->setWord($word);
```

Comment:
__optional__

Just an idea, maybe you could have a third field in menu action class - lets call it `parameters`. And the using variadic operator pass those parameters to method call. That way service would not have a changing state.